### PR TITLE
Add PublicationSelect component.

### PR DESCRIPTION
**Pull request by ankitrox**
## Summary

Addresses issue:

Note: `publicationOnboardingStateLastSyncedAtMs` is not getting updated as part of `onPublicationChange` function because it will be updated as part of the `syncPublicationOnboardingState` action, which will run immediately before (`getPublications` resolver)/after (`maybeSyncPublicationsOnboardingState`) a publication is selected.

- #8837 

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/reader-revenue-manager/components/common/PublicationSelect.test.js [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+		);
+
+		await waitForRegistry();
+
+		const listItems = getAllByRole( 'menuitem', { hidden: true } );
+		expect( listItems ).toHaveLength( 1 );
+
+		// Verify that the Publication select dropdown is disabled.
+		[
+			'.googlesitekit-analytics-4__select-publication',
```
Comment (nfmohit) [inline assets/js/modules/reader-revenue-manager/components/common/PublicationSelect.test.js:121]: This is an unrelated class name which has been proposed to be removed. Let's remove it here as well. If it is necessary to persist, let's change it to something like `.googlesitekit-rrm__select-publication`.

File: assets/js/modules/reader-revenue-manager/components/common/PublicationSelect.test.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+			'.mdc-select--disabled',
+		].forEach( ( className ) => {
+			expect( container.querySelector( className ) ).toBeInTheDocument();
+		} );
+	} );
+
+	it( 'should update publicationID in the store when a new item is selected', async () => {
```
Comment (nfmohit) [inline assets/js/modules/reader-revenue-manager/components/common/PublicationSelect.test.js:128]: Let's test the same for `publicationOnboardingState` here as well. It would be nice to use `it.each` as the test cases would be very similar.

**Review by nfmohit — CHANGES_REQUESTED** [review]
Thank you for your excellent work here, @ankitrox ! Please take a look at my feedback below:

1. Please take a look at my CR comments.
2. I understand it is not a part of the ACs/IB, but to help the QA team test the component, let's render the `PublicationSelect` component in the placeholder `SetupMain` component, i.e. `assets/js/modules/reader-revenue-manager/components/setup/SetupMain.js`.
	1. This will allow the QA team to play with the `PublicationSelect` component in the RRM setup screen, or the `SetupMain` Storybook story.
	2. We do not need to ensure here that the styling is on point. This will act as a placeholder component to assist with QA testing.
    3. Let's add a comment to mention that this is just added as a placeholder component to assist with testing.
    4. Let's update the QAB to include testing instructions.
3. I see that you have not included the update of the `publicationOnboardingStateLastSyncedAtMs` module setting inside the `onPublicationChange` function. This actually makes sense because it will be updated as part of the `syncPublicationOnboardingState` action anyway which will run almost immediately before (`getPublications` resolver)/after (`maybeSyncPublicationsOnboardingState`) a publication is selected. However, while this looks good to me, as this is a deviation from the ACs/IB, I'd suggest mentioning this and the reason in the PR description.

Please let me know if you have any questions on the above, thank you!

**Comment by ankitrox** [comment]
Thank you so much for your feedback @nfmohit and reviewing the PR.

1. I've addressed the comments in the PR.
2. Added the deviation from IB/AC in PR description.
3. Updated the QAB by adding `PublicationSelect` in `SetupMain` component.

Assigning this to you for re-review.

Thanks

File: assets/js/modules/reader-revenue-manager/components/common/PublicationSelect.test.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+			.dispatch( MODULES_READER_REVENUE_MANAGER )
+			.setPublicationID( 'ABCD*&^' );
+
+		const { container, waitForRegistry } = render( <PublicationSelect />, {
+			registry,
+		} );
+
+		await waitForRegistry();
+
+		// After we set a valid publication ID, the publication select should be visible.
```
Comment (techanvil) [inline assets/js/modules/reader-revenue-manager/components/common/PublicationSelect.test.js:63]: This test seems incomplete. For one thing, we should test for the presence of the `.mdc-select--invalid` class when the publication ID is invalid. Secondly, the test is currently not actually setting a valid publication ID, despite this comment suggesting that it has.

File: assets/js/modules/reader-revenue-manager/components/common/PublicationSelect.test.js [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+			<PublicationSelect />,
+			{
+				registry,
+			}
+		);
+
+		await waitForRegistry();
+
+		const listItems = getAllByRole( 'menuitem', { hidden: true } );
+		expect( listItems ).toHaveLength( publications.length );
```
Comment (techanvil) [inline assets/js/modules/reader-revenue-manager/components/common/PublicationSelect.test.js:78]: It would be nice to verify the options have the correct values, as well as there being the correct number of them.

File: assets/js/modules/reader-revenue-manager/components/common/PublicationSelect.js [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+				// eslint-disable-next-line sitekit/acronym-case
+				( { publicationId } ) => publicationId === newPublicationID
+			);
+
+			// eslint-disable-next-line sitekit/acronym-case
+			const { publicationId, onboardingState } = publication;
+
+			// eslint-disable-next-line sitekit/acronym-case
+			setPublicationID( publicationId );
+			setPublicationOnboardingState( onboardingState );
```
Comment (techanvil) [inline assets/js/modules/reader-revenue-manager/components/common/PublicationSelect.js:97]: This can be simplified a bit:

```suggestion
			setPublicationID( newPublicationID );
			setPublicationOnboardingState( publication.onboardingState );
```

**Review by techanvil — CHANGES_REQUESTED** [review]
Thanks @ankitrox, this is looking good. I've left a few comments, please take a look.

**Comment by ankitrox** [comment]
Thanks for reviewing this @techanvil .

I've addressed the comments added to the PR.

Send your way to have another review.

**Review by techanvil — APPROVED** [review]
Thanks @ankitrox! This LGTM.

Please note, I made an additional tweak to the `SetupMain` test to avoid the test needing to know about the underlying REST routes.

We generally prefer to avoid this, unless the tests are explicitly making assertions about the network requests.
